crates/relayer-framework/src/full/one_for_all/impls/batch.rs

```diff
@@ -91,9 +91,11 @@ where
 impl<Relay> HasBatchContext<SourceTarget> for OfaRelayWrapper<Relay>
 where
     Relay: OfaBaseRelay,
-    Relay::SrcChain: OfaFullChain,
+    // TODO: do not require the chain error to be the same as relay error.
+    // this can be fixed in #2816.
+    Relay::SrcChain: OfaFullChain<Error = Relay::Error>,
```

Comment:
This requires refactoring on the batch context, but the batch context can be removed through #2816. So we will leave it to #2816 to fix this final limitation.